src/platform/datapath_raw.c

```diff
@@ -589,32 +589,63 @@ CXPLAT_THREAD_CALLBACK(CxPlatRouteResolutionWorkerThread, Context)
         CxPlatDispatchLockRelease(&Worker->Lock);
 
         while (!CxPlatListIsEmpty(&Operations)) {
+            MIB_IPNET_ROW2 IpnetRow = {0};
             CXPLAT_ROUTE_RESOLUTION_OPERATION* Operation =
                 CXPLAT_CONTAINING_RECORD(
                     CxPlatListRemoveHead(&Operations), CXPLAT_ROUTE_RESOLUTION_OPERATION, WorkerLink);
-            NETIO_STATUS Status =
-            Status = GetIpNetEntry2(&Operation->IpnetRow);
-            if (Status != ERROR_SUCCESS || Operation->IpnetRow.State <= NlnsIncomplete) {
-                Status =
-                    ResolveIpNetEntry2(&Operation->IpnetRow, NULL);
-                if (Status != 0) {
+            CXPLAT_DBG_ASSERT(
+                Operation->Route.State == RouteRefreshing || Operation->Route.State == RouteResolving);
+            CXPLAT_ROUTE NewRoute = Operation->Route;
+            QUIC_STATUS Status = CxPlatQueryRoute(Operation->Socket, &NewRoute, &IpnetRow);
```

Comment:
Can you remind me why we now also have a call of GetBestRoute2 here in the worker thread? Why can't the call be just in CxPlatResolveRoute? I thought that was what we had discussed.

Reply:
It's cleaner for refresh case because we only need to call CxPlatQueryRoute once. We don't call CxPlatQueryRoute in ResolveRoute when we try to refresh the route. No more double-check GetIpNetEntry2. We could also skip calling CxPlatQueryRoute for initial resolution in the worker but resolving initial route in the worker is a rare case IMO.